rc/base/d.kak

```diff
@@ -116,18 +116,17 @@ define-command -hidden d-indent-on-closing-curly-brace %[
 # Initialization
 # ‾‾‾‾‾‾‾‾‾‾‾‾‾‾
 
-hook -group d-highlight global WinSetOption filetype=d %{ add-highlighter window/d ref d }
+hook -group d-highlight global WinSetOption filetype=d %{
+    add-highlighter window/d ref d
+    hook -once -always window WinSetOption filetype=(?!d).* %{ remove-highlighter window/d }
```

Comment:
Same remark as with c, this hook may not be triggered depending on the new filetype